aim/sdk/objects/figure.py

```diff
@@ -16,7 +18,9 @@ class Figure(CustomObject):
     def __init__(self, obj):
         super().__init__()
 
-        if inst_has_typename(obj, ['Figure', 'BaseFigure']):
+        if inst_has_typename(obj, ['Figure.matplotlib.figure']):
```

Comment:
@mihran113's comment applies here as well; should be `inst_has_typename(image, ['Figure', 'matplotlib', 'figure'])`